### DISCUSSION TITLE
Debugger: Delve Interface

### DISCUSSION THREAD

**Discussion by emad-elsaid** [go-delve/delve#3126] [Show and tell]
For the past couple weeks I have been working on an interface for delve. and I think it's in a good state for early testing.

https://github.com/emad-elsaid/debugger

The project uses Delve `debugger` service directly. and uses [Gio](https://gioui.org/) package for UI. similar to what @aarzilli did for [gdlv](https://github.com/aarzilli/gdlv)

My goal for this project is to have a similar features to what frontend devs have in Chrome developer tools. so it'll be probably using many other libraries to collect information about the running program.

![main](https://user-images.githubusercontent.com/54403/187047033-a5639a42-59cf-4453-9da0-7507db5da8bb.png)

**Comment by aarzilli** [comment]
If you haven't seen them already: 
* https://github.com/go-delve/delve/tree/master/Documentation/api
* https://github.com/go-delve/delve/blob/master/Documentation/api/ClientHowto.md
* https://github.com/aarzilli/delve_client_testing

And if you want you can add to the list of front ends https://github.com/go-delve/delve/blob/master/Documentation/EditorIntegration.md

**Reply by emad-elsaid** [reply]
Opened a PR to add it to the list of UIs 
https://github.com/go-delve/delve/pull/3127
